ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -0,0 +1,256 @@
+      GetRequest getRequest = createGetRequest(blobId, getOperationFlag());
+      RequestInfo request = new RequestInfo(hostname, port, getRequest);
+      int correlationId = getRequest.getCorrelationId();
+      correlationIdToGetRequestInfo.put(correlationId, new GetRequestInfo(replicaId, time.milliseconds()));
+      requestRegistrationCallback.registerRequestToSend(this, request);
+      replicaIterator.remove();
+    }
+  }
+
+  /**
+   * Handle the given {@link ResponseInfo} and update the status of the operation.
+   * Based on the status of the response, this involves handling the body, notifying the operation tracker,
+   * notifying the response handler and possibly setting the operationException.
+   * @param responseInfo the {@link ResponseInfo} to be handled.
+   */
+  @Override
+  void handleResponse(ResponseInfo responseInfo) {
+    if (isOperationComplete()) {
+      return;
+    }
+    int correlationId = ((GetRequest) responseInfo.getRequest()).getCorrelationId();
+    // Get the GetOperation that generated the request.
+    GetRequestInfo getRequestInfo = correlationIdToGetRequestInfo.remove(correlationId);
+    if (getRequestInfo == null) {
+      // Ignore right away. This associated operation has completed.
+      return;
+    }
+    if (responseInfo.getError() != null) {
+      setOperationException(new RouterException("Operation timed out", RouterErrorCode.OperationTimedOut));
+      responseHandler.onRequestResponseException(getRequestInfo.replicaId, new IOException("NetworkClient error"));
+      operationTracker.onResponse(getRequestInfo.replicaId, false);
+    } else {
+      try {
+        GetResponse getResponse = GetResponse
+            .readFrom(new DataInputStream(new ByteBufferInputStream(responseInfo.getResponse())), clusterMap);
+        if (getResponse.getCorrelationId() != correlationId) {
+          // The NetworkClient associates a response with a request based on the fact that only one request is sent
+          // out over a connection id, and the response received on a connection id must be for the latest request
+          // sent over it. The check here ensures that is indeed the case. If not, log an error and fail this request.
+          // There is no other way to handle it.
+          setOperationException(
+              new RouterException("The correlation id in the GetResponse " + getResponse.getCorrelationId() +
+                  "is not the same as the correlation id in the associated GetRequest: " + correlationId,
+                  RouterErrorCode.UnexpectedInternalError));
+          operationTracker.onResponse(getRequestInfo.replicaId, false);
+          // we do not notify the ResponseHandler responsible for failure detection as this is an unexpected error.
+        } else {
+          processGetResponse(getRequestInfo, getResponse);
+        }
+      } catch (IOException | MessageFormatException e) {
```

Comment:
I don't think we support 1.6 - you could build and check. I think certain changes introduced for SSL only work with 1.7.